libraries/api-extractor-model/src/mixins/ApiItemContainerMixin.ts

```diff
@@ -143,7 +146,7 @@ export function ApiItemContainerMixin<TBaseClass extends IApiItemConstructor>(
 
     /** @override */
     public get members(): ReadonlyArray<ApiItem> {
-      if (!this[_membersSorted]) {
+      if (!this[_membersSorted] && !this.preserveEnumMemberOrder) {
```

Comment:
I _think_ on deserialization we always want to set `options.shouldSortMembers` to `false` (which we're not doing right now). Because whatever order is in the `.api.json` is the one we want to preserve upon deserialization. Otherwise, if you serialized an `.api.json` in source order, and then deserialized it, I think subsequent calls to `members` would sort them again.

Thus, I think if you added an api-documenter test, what you have so far here wouldn't work. Am I missing something though?